### ISSUE TITLE
Current version of ERC721 token is not compatible with etherscan

### ISSUE THREAD

**Issue by igorline** [vyperlang/vyper#1575]
I've tried deploying current version of ERC721 token contract in this repository in rinkeby network and minted one token and etherscan is actually showing that my token is ERC20.
https://rinkeby.etherscan.io/token/0x923844f7b3fd4f8a30f5b26002efd420dba49037
This way you are not able to see any data on holders, total supply, etc..

I previously tried deploying it to the mainnet and it was the same issue.
Though I was able to fix it by adding staightforward method in the contract found somewhere on github.

```
@public
@constant
def implementsERC721() -> bool:
  return True
```

Not sure if the issue is related to vyper or etherscan itself.

**Comment by fubuloubu** [comment]
Coming back to this from another place, but `ERC721` does not specify a `implementsERC721` function. So likely something that's wrong with Etherscan.

`ERC721` specifies `ERC165` as a mandatory dependency, which could be where this issue comes from

**Comment by charles-cooper** [comment]
@fubuloubu is this still an issue?

**Comment by fubuloubu** [comment]
Closed as of #2709
